homework/index.js

```diff
@@ -29,23 +31,117 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function addRow(table, labelText, value) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, { text: `${labelText}:` });
+    createAndAppend('td', tr, { text: value });
+    return tr;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+  function renderRepoDetails(repo, div) {
+    div.innerHTML = ''; // to clear result of the previous selection
+    const table = createAndAppend('table', div, { class: 'repos-table' });
+    const firstRow = addRow(table, 'Name', '');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    addRow(table, 'Description', repo.description);
+    addRow(table, 'Forks', repo.forks);
+    addRow(
+      table,
+      'Updated at',
+      new Date(repo.updated_at).toLocaleString('en-GB', { timeZone: 'UTC' }),
+    );
+  }
+
+  /* must declare variable: root, before the next function. */
+  const root = document.getElementById('root');
+
+  function renderContributors(repo, ul) {
+    fetchJSON(repo.contributors_url)
+      .then(contributors => {
+        ul.innerHTML = '';
+        createAndAppend('p', ul, {
+          text: `Contributors:`,
+          id: 'contributors-title',
+        });
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul);
+          const table = createAndAppend('table', li, {
```

Comment:
This is a table with a single row. It works, but a would a couple of divs work here too? It would give you more CSS styling flexibility.

Reply:
And another one. I would not use a table here, but I'm fine with you leaving it like it is.

Reply:
Oh, I see. 
I have replaced the contributors table with divs.